runner/scheduler.py

```diff
@@ -475,6 +459,7 @@ def update_result(
                 result.max_score = max_score
                 result.reasoning = reason
 
+        result.needles = len(example.script) - example.number_of_questions
```

Comment:
repeated from line 318

Reply:
I will keep this one. The update here is needed for more dynamic tests like restaurant and jokes.